File: hw/sivinta/U3/HW16MazeSolid/MazePython/model/room.py

from typing import List
from model.door import Door

class Room:
    def __init__(self, x: int, y: int):
        self._x = x
        self._y = y
        self._doors: List[Door] = []

    @property
    def x(self) -> int:
        return self._x

    @property
    def y(self) -> int:
        return self._y

    @property
    def doors(self) -> List[Door]:
        return self._doors

    def add_door(self, door: Door) -> None:
        self._doors.append(door)

File: hw/maisincho/u3/HW17MazeSolid/MazeJava/src/main/java/ec/edu/espe/mazegenerator/model/Room.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.mazegenerator.model;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Isaac Maisincho Crafters_Market DCCO ESPE
 */
public class Room {

    private final int x;
    private final int y;
    private final List<Door> doors;

    public Room(int x, int y) {
        this.x = x;
        this.y = y;
        this.doors = new ArrayList<>();
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public List<Door> getDoors() {
        return doors;
    }

    public void addDoor(Door door) {
        doors.add(door);
    }
}
